shashi/loops.js

```diff
@@ -0,0 +1,222 @@
+var arr = [1,2,3,4,5];
+
+//all odd
+function allOdd(array) {
+  var newArrOdd = [];
+  for( var i=0;i<array.length;i++){
+      if((array[i]%2) == 1) {
+        newArrOdd.push(array[i])
+      }
+  }
+  return newArrOdd;
+} 
+
+function allOddWhile(array){
+  var newArrOdd = [];
+  var i=0;
+  while(i<array.length){
+    if((array[i]%2) == 1) {
+      newArrOdd.push(array[i])
+    }
+    i++;
+  }
+  return newArrOdd;
+}
+allOdd(arr);
+allOddWhile(arr);
+
+//all even
+function allEven(array) {
+  var newArrEven = [];
+  for( var i=0;i<array.length;i++){
+      if((array[i]%2) != 1) {
+        newArrEven.push(array[i])
+      }
+  }
+  return newArrEven;
+} 
+
+function allEvenWhile(array){
+  var newArrEven = [];
+  var i=0;
+  while(i<array.length){
+    if((array[i]%2) != 1) {
+      newArrEven.push(array[i])
+    }
+    i++;
+  }
+  return newArrEven;
+}
+
+allEven(arr);
+allEvenWhile(arr);
+
+//backwords
+function backwords(array){
+  var backWordArr = [];
+  for(var i=array.length-1; i>=0; i--){
+    backWordArr.push(array[i]);
+  }
+  return backWordArr;
+}
+
+function backwordsWhile(array){
+  var backWordArr = [];
+  var i = array.length-1;
+  while(i>=0){
+    backWordArr.push(array[i]);
+    i--;
+    }
+  return backWordArr;
+}
+
+backwords(arr);
+backwordsWhile(arr);
+
+var arr1 = [6,7,8,9,10];
+var arr = [1,2,3,4,5];
+
+//combine both arrays
+
+function combineArr(array1,array2){
+  var combinedArr =[];
+   combinedArr= array1;
```

Comment:
This does not create a new array for combinedArr. In console, try this and push to combinedArr and then see what the value of array1 is.